lib/components/src/Zoom/ZoomElement.tsx

```diff
@@ -24,15 +62,28 @@ export function ZoomElement({ scale, children }: ZoomProps) {
   const componentWrapperRef = useRef<HTMLDivElement>(null);
   const [height, setHeight] = useState(0);
 
+  const handleMutations = useCallback(() => {
+    setHeight(componentWrapperRef.current.getBoundingClientRect().height);
+  }, []);
+
   useEffect(() => {
     if (componentWrapperRef.current) {
       setHeight(componentWrapperRef.current.getBoundingClientRect().height);
     }
   }, [scale, componentWrapperRef.current]);
 
+  useMutationObserver({
+    element: componentWrapperRef,
+    options: { subtree: true, childList: true },
+    callback: handleMutations,
+  });
+
   return (
     <ZoomElementWrapper scale={scale} height={height}>
-      <div ref={componentWrapperRef} className="innerZoomElementWrapper">
+      <div
+        ref={browserSupportsCssZoom() ? null : componentWrapperRef}
```

Comment:
I think we can just call `browserSupportsCssZoom` once, outside the component render function, put it in a variable and use that here. Otherwise we'll be running this check over and over again.